tests/functional/test_user_icon_click.py

```diff
@@ -0,0 +1,41 @@
+"""
+Functional test for logging out of the SecureDrop client application. The test
+is based upon the client testing descriptions here:
+
+https://github.com/freedomofpress/securedrop-client/wiki/Test-plan#basic-client-testing
+"""
+import pytest
+import time
+import pyautogui
+
+from .utils import get_safe_tempdir, get_logged_in_test_context
+
+
+@pytest.mark.vcr()
+def test_user_icon_click(qtbot, mocker):
+    """
+    WARNING: THIS TEST CAUSES SUBSEQUENT TESTS TO CRASH WITH A CORE DUMP!
+
+    As a result it should be run in isolation (see the test-functional section
+    of the Makefile for details). Why does it crash? I suspect shared state
+    leaking via the qtbot instance passed into tests.
+
+    A journalist can successfully see the logout menu by clicking the user icon.
+    """
+    totp = "079978"
+    tempdir = get_safe_tempdir()
+    gui, controller = get_logged_in_test_context(tempdir, qtbot, totp)
+
+    def check_login_button():
+        assert gui.left_pane.user_profile.login_button.isVisible()
+
+    qtbot.wait(5000)
+    # Now instead of clicking via qtbot, we can click via mouse
+    pyautogui.click(25, 70)
```

Comment:
maintainability comment: we have an increasing number of magic numbers in our functional tests - we have in here three timeouts and two (x,y) pixel positions to click. For the former, we could start making these constants and putting in a common file so it's clear why they are the length they are, e.g. `TIME_TO_RENDER_SOURCE_LIST` (or whatever the reason is). For the latter, I realize that was done due to https://github.com/freedomofpress/securedrop-client/pull/1022#issuecomment-605104048 - could we do something like use [`QWidget.mapToGlobal`](https://doc.qt.io/qt-5/qwidget.html#mapToGlobal) to get these dynamically?

Reply:
For the timeouts, we can do that in a separate change - I have a wip branch `functest-magic-numbers` where I have some wip in that direction. For merge of this PR, let's just look at the `(x, y)` pixel positions - what do you think about my suggestion above regarding getting the positions dynamically @kushaldas?

Reply:
(made PR #1086 for the first part)

Reply:
See commit https://github.com/freedomofpress/securedrop-client/commit/e5745e3f933eb37c67e8cfebe180c53973375c31#diff-78ab0aa7d821271795a6e70102e15f38R44 - the menu opens here for me without specifying pixel positions - can you see if it does for you?

For the next call to `pyautogui.click` on line 39, you'll need to implement the TODO inline [here](https://github.com/freedomofpress/securedrop-client/commit/e5745e3f933eb37c67e8cfebe180c53973375c31#diff-78ab0aa7d821271795a6e70102e15f38R46-R48), basically to adjust the position returned [here](https://github.com/freedomofpress/securedrop-client/commit/e5745e3f933eb37c67e8cfebe180c53973375c31#diff-78ab0aa7d821271795a6e70102e15f38R44) where you expect the menu to appear, I think that will work. Want to give that a try? I can take another look tomorrow based on your findings :-)

The nice thing about this is if that works then I think we won't need the hardcoded `1680x1050x24` screen size anymore and thus the Linux-only concern will go away too.

Reply:
It did not show the menu in my linux box. I will try to record a GIF to show you.

Reply:
hmm odd, thanks for checking, I'll take a look

Reply:
check out commit 6a7f98f84d931197fa243c6d168538c716799012 - this is a cleaned up version pushed to this branch, now passing in CI (Linux) and for me locally (macOS), let me know what you think.